app/repositories/TodoRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\Todo;
+use App\Entities\TodoEntity;
+use Illuminate\Auth\AuthManager;
+
+
+class TodoRepository
+{
+    private AuthManager $authService;
+    public function __construct(AuthManager $authService)
```

Comment:
Repository classes should not care about the auth. you should do it in The Service classes. then pass it to the TodoEntity